infra/Pos/Diffusion/Subscription/Dns.hs

```diff
@@ -74,25 +72,28 @@ dnsSubscriptionWorker oq networkCfg DnsDomains{..} keepaliveTimer nextSlotDurati
     -- (see 'retryInterval').
     subscribeAlts
         :: SharedAtomicT m (Map Int (Alts NodeId))
-        -> MVar Millisecond
+        -> Maybe (MVar Millisecond)
         -> (Int, Alts (NodeAddr DNS.Domain))
         -> m ()
     subscribeAlts _ _ (index, []) =
         logWarning $ sformat ("dnsSubscriptionWorker: no alternatives given for index "%int) index
-    subscribeAlts dnsPeersVar subDuration (index, alts) = do
+    subscribeAlts dnsPeersVar mSubDuration (index, alts) = do
         -- Any DNSError is squelched. So are IOExceptions, for good measure.
         -- This does not include async exceptions.
         -- It does handle the case in which there's no internet connection, or
         -- a bad configuration, so that the subscription thread will keep on
         -- retrying.
+        subDuration <- case mSubDuration of
+            Just sd -> return sd
+            Nothing -> newMVar (0 :: Millisecond)
```

Comment:
@avieth if you think creating a single mutable cell for each list of alternatives will allocate too much memory, we can do this the other way around, i.e. reset the `subDuration` when all threads have run at least once. This would requires to store a list of `[Bool]` (hence it will only be lineary more memory efficient).